### PR TITLE
feat: engine type implementation (will replace node api preview feature)

### PR DESCRIPTION
**Pull request by williamluke4**

### CODE REVIEW

File: platforms-serverless/firebase-functions/functions/prisma/schema-with-binary.prisma [hunk 1]
```diff
@@ -1,7 +1,7 @@
 generator client {
   provider      = "prisma-client-js"
   binaryTargets = ["native", "debian-openssl-1.1.x"]
-  previewFeatures = ["nApi"]
+  engineType = "binary"
```
Comment (janpio) [inline platforms-serverless/firebase-functions/functions/prisma/schema-with-binary.prisma:4]:
```suggestion
  engineType    = "binary"
```

File: platforms-serverless/vercel-with-redwood/web/package.json [hunk 2]
```diff
@@ -13,11 +13,15 @@
...
     "prop-types": "15.7.2",
     "react": "17.0.2",
     "react-dom": "17.0.2"
+  },
+  "devDependencies": {
+    "prisma": "2.29.0-integration-enabling-node-api.4"
   }
-}
\ No newline at end of file
+}
```
Comment (janpio) [inline platforms-serverless/vercel-with-redwood/web/package.json]: why is this necessary? I do not understand this redwood project

File: .github/scripts/check-cli-binaries.sh [hunk 3]
```diff
@@ -36,8 +48,8 @@ case $(uname | tr '[:upper:]' '[:lower:]') in
     ;;
 esac
 echo "Assumed OS: $os_name"
-
-if [ -z ${PRISMA_FORCE_NAPI+x} ]; then
+echo $CLI_QUERY_ENGINE_TYPE == "binary"
+if [ $CLI_QUERY_ENGINE_TYPE == "binary" ]; then
   echo "Node-API: Disabled"
```
Comment (Jolg42) [inline .github/scripts/check-cli-binaries.sh:53]: Maybe renaming these debug logs would make more sense?

File: .github/workflows/optional-test.yaml [hunk 4]
```diff
@@ -67,7 +67,7 @@ jobs:
             prisma-json-schema-generator,
             prisma-nestjs-graphql,
           ]
-        engine: ['binary'] #['node-api', 'binary']
+        clientEngine: ['node-api'] #['node-api', 'binary']
```
Comment (Jolg42) [inline .github/workflows/optional-test.yaml]: Why the switch / only testing node-api now?

File: .github/workflows/test.yaml [hunk 5]
```diff
@@ -92,7 +88,7 @@ jobs:
       fail-fast: false
       matrix:
         feature: [pm2]
-        engine: [binary] # ['node-api', 'binary']
+        clientEngine: [node-api] # ['node-api', 'binary']
```
Comment (Jolg42) [inline .github/workflows/test.yaml]: Why the switch / only testing node-api now?

File: .github/workflows/test.yaml [hunk 6]
```diff
@@ -434,7 +424,7 @@ jobs:
         framework:
           - nestjs
           - nextjs
-        engine: [binary] #['node-api', 'binary']
+        clientEngine: [node-api] #['node-api', 'binary']
```
Comment (Jolg42) [inline .github/workflows/test.yaml]: Why the switch / only testing node-api now?

File: .github/workflows/test.yaml [hunk 7]
```diff
@@ -781,7 +629,7 @@ jobs:
           - webpack
           - parcel
           - rollup
-        engine: [binary] #['node-api', 'binary']
+        clientEngine: [node-api] #['node-api', 'binary']
```
Comment (Jolg42) [inline .github/workflows/test.yaml]: Why the switch / only testing node-api now?

File: .github/workflows/test.yaml [hunk 8]
```diff
@@ -831,7 +678,7 @@ jobs:
           - apollo-server
           - type-graphql
           - nexus-schema
-        engine: [binary] #['node-api', 'binary']
+        clientEngine: [node-api] #['node-api', 'binary']
```
Comment (Jolg42) [inline .github/workflows/test.yaml]: Why the switch / only testing node-api now?

File: .github/workflows/test.yaml [hunk 9]
```diff
@@ -890,7 +736,7 @@ jobs:
           - supabase
           - supabase-pool
           - planetscale
-        engine: [binary] #['node-api', 'binary']
+        clientEngine: [node-api] #['node-api', 'binary']
```
Comment (Jolg42) [inline .github/workflows/test.yaml]: Why the switch / only testing node-api now?

File: .github/workflows/test.yaml [hunk 10]
```diff
@@ -952,7 +797,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        engine: [binary] # ['node-api', 'binary']
+        clientEngine: [node-api] # ['node-api', 'binary']
```
Comment (Jolg42) [inline .github/workflows/test.yaml]: Why the switch / only testing node-api now?

File: .github/scripts/check-cli-binaries.sh (hunk 3 above)
Comment (janpio) [reply .github/scripts/check-cli-binaries.sh:53]: Rename to what and why?

File: .github/workflows/optional-test.yaml (hunk 4 above)
Comment (janpio) [reply .github/workflows/optional-test.yaml]: This PR switched the default engine type from `binary` to `node-api`, so all the tests that only test one of these (for capacity reasons), also were switched over.

Now that this is about the binary as default again, this indeed does not make sense.

File: .github/workflows/test.yaml (hunk 5 above)
Comment (janpio) [reply .github/workflows/test.yaml]: https://github.com/prisma/e2e-tests/pull/1954#discussion_r684442790

File: .github/workflows/test.yaml (hunk 6 above)
Comment (janpio) [reply .github/workflows/test.yaml]: https://github.com/prisma/e2e-tests/pull/1954#discussion_r684442790

File: .github/workflows/test.yaml (hunk 7 above)
Comment (janpio) [reply .github/workflows/test.yaml:632]: https://github.com/prisma/e2e-tests/pull/1954#discussion_r684442790

File: .github/workflows/test.yaml (hunk 8 above)
Comment (janpio) [reply .github/workflows/test.yaml]: https://github.com/prisma/e2e-tests/pull/1954#discussion_r684442790

File: .github/workflows/test.yaml (hunk 9 above)
Comment (janpio) [reply .github/workflows/test.yaml]: https://github.com/prisma/e2e-tests/pull/1954#discussion_r684442790

File: .github/workflows/test.yaml (hunk 10 above)
Comment (janpio) [reply .github/workflows/test.yaml]: https://github.com/prisma/e2e-tests/pull/1954#discussion_r684442790

**Review by janpio — COMMENTED** [review]
- "Checking Binaries" scripts does not make sense for `engine-types` tests I think
- `engine-types` are really quite impossible to review from the output we get from the running tests :(

File: .github/workflows/optional-test.yaml (hunk 4 above)
Comment (williamluke4) [reply .github/workflows/optional-test.yaml]: Changed back to `['binary']`

File: .github/workflows/test.yaml (hunk 5 above)
Comment (williamluke4) [reply .github/workflows/test.yaml]: Changed back to `['binary']`

File: .github/workflows/test.yaml (hunk 6 above)
Comment (williamluke4) [reply .github/workflows/test.yaml]: Changed back to `['binary']`

File: .github/workflows/test.yaml (hunk 8 above)
Comment (williamluke4) [reply .github/workflows/test.yaml]: Changed back to `['binary']`

File: .github/workflows/test.yaml (hunk 10 above)
Comment (williamluke4) [reply .github/workflows/test.yaml]: Changed back to `['binary']`

File: .github/workflows/test.yaml (hunk 9 above)
Comment (williamluke4) [reply .github/workflows/test.yaml]: Changed back to `['binary']`

File: .github/workflows/test.yaml (hunk 7 above)
Comment (williamluke4) [reply .github/workflows/test.yaml]: Changed back to `['binary']`
